In [1]:
import pm_evaluation

### Evaluation

We use gt for evaluations. We do blocking and pw on gt.

#### Blocking

In [2]:
import importlib.util
import sys

module_dir = "../pairwise_matching/ditto"
module_path = "../pairwise_matching/ditto/matcher.py"

# Add the directory to sys.path (to allow imports inside the module to work)
if module_dir not in sys.path:
    sys.path.append(module_dir)

# Load the module dynamically
module_name = "your_module_name"
spec = importlib.util.spec_from_file_location(module_name, module_path)
ditto_matcher = importlib.util.module_from_spec(spec)
spec.loader.exec_module(ditto_matcher)

module_dir = "../pairwise_matching"
module_path = "../pairwise_matching/rll_pairwise_matching.py"

# Add the directory to sys.path (to allow imports inside the module to work)
if module_dir not in sys.path:
    sys.path.append(module_dir)

# Load the module dynamically
module_name = "your_module_name"
spec = importlib.util.spec_from_file_location(module_name, module_path)
jw_matcher = importlib.util.module_from_spec(spec)
spec.loader.exec_module(jw_matcher)

' module_dir = "../pairwise_matching"\nmodule_path = "../pairwise_matching/rll_pairwise_matching.py"\n\n# Add the directory to sys.path (to allow imports inside the module to work)\nif module_dir not in sys.path:\n    sys.path.append(module_dir)\n\n# Load the module dynamically\nmodule_name = "your_module_name"\nspec = importlib.util.spec_from_file_location(module_name, module_path)\njw_matcher = importlib.util.module_from_spec(spec)\nspec.loader.exec_module(jw_matcher) '

In [3]:
blocking_paths = ["lsh_bigram_gt_blocking.json", "lsh_words_gt_blocking.json", "lsh_bigram_aq_gt_blocking.json", "lsh_words_aq_gt_blocking.json" ]
pairwise_methods = ["jarowinkler", "ditto", "deepmatcher"]

for blocking_path in blocking_paths:
        pairs: list = pm_evaluation.get_pairs_for_pairwise_matching(blocking_path)
        print(len(pairs))
        
        for pw_method in pairwise_methods:
            if pw_method == "jarowinkler":
                for threshold in [0.7, 0.75]:
                    # do jarowinkler pw and save to file
                    jw_matcher.pairwise_matching(blocking_path, threshold)
            if pw_method == "ditto":
                blocking_path = blocking_path.replace("json", "txt")
                ditto_matcher.pairwise_matching(pairs_to_evaluate=pairs,
                                                checkpoint_path='../pairwise_matching/ditto/checkpoints/',
                                                ditto_path='../pairwise_matching/ditto/',
                                                output_path='results/ditto/' + blocking_path,
                                                use_gpu=True, gt_eval=True)
            elif pw_method == "deepmatcher":
                # do dm pw aoooo
                pass

609295
USING CUDAAA


Some weights of the model checkpoint at distilbert-base-uncased were not used when initializing DistilBertModel: ['vocab_transform.bias', 'vocab_layer_norm.weight', 'vocab_projector.weight', 'vocab_projector.bias', 'vocab_transform.weight', 'vocab_layer_norm.bias']
- This IS expected if you are initializing DistilBertModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing DistilBertModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
106it [00:00, 212015.37it/s]


load_f1 = 0.9465648854961832
real_f1 = 0.9465648854961832
106805
USING CUDAAA


Some weights of the model checkpoint at distilbert-base-uncased were not used when initializing DistilBertModel: ['vocab_transform.bias', 'vocab_layer_norm.weight', 'vocab_projector.weight', 'vocab_projector.bias', 'vocab_transform.weight', 'vocab_layer_norm.bias']
- This IS expected if you are initializing DistilBertModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing DistilBertModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
106it [00:00, 211813.35it/s]


load_f1 = 0.9465648854961832
real_f1 = 0.9465648854961832


#### Eval of pairwise

In [4]:
import os
RED = "\033[31m"
RESET = "\033[0m"

for dirpath, _, filenames in os.walk("results"):
    if dirpath != "results":
        name = dirpath.split("\\")[1].upper()
        print(f"{RED}{name}{RESET}")
    for filename in filenames:
        filepath = os.path.join(dirpath, filename)
        pm_evaluation.evaluate("../ground_truth/gt.txt", filepath)
        

DEEPMATCHER


TypeError: unhashable type: 'list'

#### Execution Times

##### Ditto
Time for ditto pairwise matching on 65.895 items - 609295 pairs after lsh_bigram_blocking: **71 minutes.**

Time for ditto pairwise matching on gt items with both blocking methods: **Average of 1.6 seconds**

##### Jarowinkler
Time for jarowinkler pairwise matching on 65.895 items after blocking: **Average of  25 seconds**